chainer/link.py

```diff
@@ -552,35 +646,57 @@ def __getitem__(self, name):
         """Equivalent to getattr."""
         return getattr(self, name)
 
+    def __setattr__(self, name, value):
```

Comment:
Differently from `Link.add_link`, `__setattr__` allows duplicate registration of single link to multiple parent links, although the name of the child link is overwritten. Do you intentionally do that, or do you simply forget it? I thought it might be a specification because there is an issue (#2052) addressing this problem.